### PR TITLE
Dev 67062 - Make lateral menu over content so it does not squeeze content

### PR DESCRIPTION
**Pull request by sebastian-ew**
## Description

Some code removed that was causing the sidebars to squeeze the content, changing them to overlap the content.

## Related ticket

https://3.basecamp.com/3451303/buckets/37454340/todos/7598541447

> Make sure to read the *Pull requests process* in CONTRIBUTING.md.
> 
> Main points:
> 
> * The assigned reviewer(s) must always *submit* a review (not simply add a
>   comment).
>
> * If the PR is "Approved", the creator of the PR must then:
>   * Implement the changes in the reviewer's comments, if applicable, and mark
>     them as "Resolved".
>   * Squash and merge the branch.
>   * Delete the branch.
> 
> * If the PR is "Request Changes", the creator of the PR must then:
>   * Update the code as requested.
>   * Re-request a review from the reviewer.
>   * NOT mark the comments as "Resolved", the reviewer will.
> 
> (You can keep or remove this block, as you wish.)

### CODE REVIEW

File: fw-child/resources/js/child-functions.js [hunk 1]
```diff
@@ -530,4 +517,19 @@ const $ = jQuery;
 
     $('body').removeClass('spinner-on');
   });
+
```
Comment (domcode-ew) [inline fw-child/resources/js/child-functions.js:614]: Why did you move this code out of the Vendor section where it was?

File: fw-child/resources/js/child-functions.js [hunk 2]
```diff
@@ -530,4 +517,19 @@ const $ = jQuery;
...
+  window.addEventListener('load', function () {
```
Comment (domcode-ew) [inline fw-child/resources/js/child-functions.js:615]: Is this essential? Was it not working before?

Please stick to jQuery for consistency. Use $(window).on('load', function () {})

File: fw-child/template/map/controls/data-variable.php [hunk 3]
```diff
@@ -119,12 +119,12 @@ class="fw-query-filter dropdown"
 					</div>
 
 					<div id="" class="query-container ">
```
Comment (domcode-ew) [inline fw-child/template/map/controls/data-variable.php:121]: Is this empty id necessary?

File: fw-child/template/learn/query.php [hunk 4]
```diff
@@ -22,72 +22,69 @@
...
-								'field' => 'slug',
-								'terms' => array ( $topic->slug )
-							)
+		<div class="pt-6">
+
+			<ul id="sort-menu" class="fw-query-sort list-unstyled d-flex all-caps">
+				<li class="selected me-4" data-sort="title_asc"><i class="fas fa-arrow-up me-2"></i> <?php _e ( 'Title (A–Z)', 'cdc' ); ?></li>
+				<li class="me-4" data-sort="title_desc"><i class="fas fa-arrow-down me-2"></i> <?php _e ( 'Title (Z–A)', 'cdc' ); ?></li>
+				<li class="me-4" data-sort="date_desc"><i class="fas fa-arrow-up me-2"></i> <?php _e ( 'Newest First', 'cdc' ); ?></li>
+				<li class="" data-sort="date_asc"><i class="fas fa-arrow-down me-2"></i> <?php _e ( 'Oldest First', 'cdc' ); ?></li>
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/learn/query.php]: Fixed!

File: fw-child/template/learn/query.php [hunk 5]
```diff
@@ -22,72 +22,69 @@
...
+			<h4 class="learn-topic-title mb-7 d-flex align-items-center font-weight-normal">
+				<span class="circle-number"><?php echo $i; ?></span>
+				<?php
+
+					echo fw_get_field ( 'title', 'topic_' . $topic->term_id );
+
+				?>
+			</h4>
+
+			<div id="" class="query-container ">
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/learn/query.php]: Fixed!

File: fw-child/template/news/query.php [hunk 6]
```diff
@@ -22,47 +22,44 @@
...
-				);
-				
-			?>
-			
-			<div id="news-grid" class="py-6" data-args='<?php echo json_encode ( $module_args ); ?>'>
+		<div class="pt-6">
+
+			<ul id="sort-menu" class="fw-query-sort list-unstyled d-flex all-caps">
+				<li class="selected me-4" data-sort="date_desc"><i class="fas fa-arrow-up me-2"></i> <?php _e ( 'Newest First', 'cdc' ); ?></li>
+				<li class="" data-sort="date_asc"><i class="fas fa-arrow-down me-2"></i> <?php _e ( 'Oldest First', 'cdc' ); ?></li>
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/news/query.php]: Fixed!

File: fw-child/template/news/query.php [hunk 7]
```diff
@@ -22,47 +22,44 @@
...
 
-						<div class="fw-query-item"></div>
+		?>
 
-					</div>
+		<div id="news-grid" class="py-6" data-args='<?php echo json_encode ( $module_args ); ?>'>
 
-					<div class="fw-query-pagination d-flex justify-content-between align-items-start mt-4">
-						<div class="fw-query-pagination-btn previous btn btn-secondary me-3"><?php _e( 'Previous', 'cdc' ); ?></div>
+			<div id="" class="query-container ">
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/news/query.php]: Fixed!

File: fw-child/template/variable/query.php [hunk 8]
```diff
@@ -22,40 +22,37 @@
...
-					<div class="fw-query-items row row-cols-1 row-cols-sm-2 row-cols-md-3 g-3 g-lg-6" data-options='<?php echo json_encode ( $item_options ); ?>'>
-						
-						<div class="fw-query-item"></div>
-					
-					</div>
+		<div class="pt-6">
+
+			<ul id="sort-menu" class="fw-query-sort list-unstyled d-flex all-caps">
+				<li class="selected me-4" data-sort="date_desc"><i class="fas fa-arrow-up me-2"></i> <?php _e ( 'Newest First', 'cdc' ); ?></li>
+				<li class="" data-sort="date_asc"><i class="fas fa-arrow-down me-2"></i> <?php _e ( 'Oldest First', 'cdc' ); ?></li>
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/variable/query.php]: Fixed!

File: fw-child/template/variable/query.php [hunk 9]
```diff
@@ -22,40 +22,37 @@
...
+				'post_status' => 'publish',
+				'orderby' => 'title',
+				'order' => 'asc'
+			);
+
+		?>
+
+		<div id="variable-grid" class="py-6" data-args='<?php echo json_encode ( $module_args ); ?>'>
+
+			<div id="" class="query-container ">
```
[the remark this answers is not in the document]
Comment (sebastian-ew) [reply fw-child/template/variable/query.php]: Fixed!

File: fw-child/resources/js/child-functions.js (hunk 1 above)
Comment (sebastian-ew) [reply fw-child/resources/js/child-functions.js:645]: I restored the code to the place where it was.

File: fw-child/resources/js/child-functions.js (hunk 2 above)
Comment (sebastian-ew) [reply fw-child/resources/js/child-functions.js]: This piece of code was refactored and placed again on the vendor section. It no longer relies on the window.load event.

File: fw-child/template/map/controls/data-variable.php (hunk 3 above)
Comment (sebastian-ew) [reply fw-child/template/map/controls/data-variable.php:121]: This id was removed.

File: fw-child/resources/js/child-functions.js [hunk 10]
```diff
@@ -233,19 +233,37 @@ const $ = jQuery;
...
     if ($('.query-page #control-bar-tabs').length) {
-      new $.Zebra_Pin($('#control-bar-tabs'), {
+      pinned_item = new $.Zebra_Pin($('#control-bar-tabs'), {
         top_spacing: scroll_offset,
-        contained: true,
+        contain: true,
       });
     }
 
+    let resize_observer = new ResizeObserver( function() {
```
Comment (domcode-ew) [inline fw-child/resources/js/child-functions.js:250]: I think this whole JS can be optimized.
I think you don't need the ResizeObserver, because you are already using $(window).on('resize'... and the #control-bar will not get resized if the window is not resized. Am I right?

You are checking if pinned_item != null in the ResizeObserver function but not in the window resize.
Wrap your repetitive code into small functions to avoid the repetition.

Please use camel case instead of snake case for code consistency.

**Review by domcode-ew — CHANGES_REQUESTED** [review]

File: fw-child/resources/js/child-functions.js (hunk 10 above)
Comment (sebastian-ew) [reply fw-child/resources/js/child-functions.js]: @domcode-ew the code was refactored to use only the ResizeObserver based on the conversation we had. Can you check again? Thanks!

**Review by domcode-ew — APPROVED** [review]

File: fw-child/template/apps/query.php [hunk 11]
```diff
@@ -21,39 +21,36 @@
...
+				'post_type' => 'app',
+				'post_parent' => 0,
+				'post_status' => 'publish'
+			);
+			
+		?>
+		
+		<div id="apps-grid" class="py-6" data-args='<?php echo json_encode ( $module_args ); ?>'>
+			
+			<div id="" class="query-container ">
```
Comment (xfrenette-crim) [inline fw-child/template/apps/query.php:47]:
```suggestion
			<div class="query-container">
```

File: fw-child/template/learn/query.php [hunk 12]
```diff
@@ -76,5 +75,12 @@ class="learn-topic-grid py-7"
 				<p class="alert alert-warning"><?php _e ( 'No items found.', 'cdc' ); ?></p>
 			</div>
 		</div>
+
+		<?php
+
+				$i++;
+			}
+
+		?>
```
Comment (xfrenette-crim) [inline fw-child/template/learn/query.php:84]: This code seems to be there in error (there is no `$i` variable, and there is an extra `}`)

File: fw-child/template/learn/query.php [hunk 13]
```diff
@@ -16,9 +16,8 @@
 		'item_class' => 'col'
 	);
 	?>
-
-	<div class="col-13 offset-1">
-		<div class="tab-drawer-bumper">
+	
+	<div class="col-12 offset-1 col-sm-13">
```
Comment (xfrenette-crim) [inline fw-child/template/learn/query.php:20]: A `<div>` was removed here, but it's closing `</div>` wasn't remove.

File: fw-child/template/map/controls/data-variable.php (hunk 3 above)
Comment (xfrenette-crim) [inline fw-child/template/map/controls/data-variable.php:121]:
```suggestion
					<div class="query-container">
```

File: fw-child/template/news/query.php [hunk 14]
```diff
@@ -1,57 +1,59 @@
...
-
-						<div class="fw-query-pagination-btn next btn btn-secondary"><?php _e( 'Next', 'cdc' ); ?></div>
-					</div>
+
+	<div class="col-12 offset-1 col-sm-13">
+		<div class="pt-6">
+			<ul id="sort-menu" class="fw-query-sort list-unstyled d-flex all-caps">
+				<li class="selected me-4" data-sort="date_desc"><i class="fas fa-arrow-up me-2"></i> <?php _e( 'Newest First', 'cdc' ); ?></li>
+				<li data-sort="date_asc"><i class="fas fa-arrow-down me-2"></i> <?php _e( 'Oldest First', 'cdc' ); ?></li>
+			</ul>
```
Comment (xfrenette-crim) [inline fw-child/template/news/query.php:36]: The "Newest" and "Oldest" sorting buttons have recently been removed. They should be removed here.

File: fw-child/template/variable/query.php [hunk 15]
```diff
@@ -36,51 +36,48 @@
...
 
-			<div id="variable-grid" class="py-6" data-args='<?php echo json_encode ( $variables_args ); ?>'>
-				
-				<div id="" class="query-container ">
-					<div class="fw-query-items row row-cols-1 row-cols-sm-2 row-cols-md-3 g-3 g-lg-6" data-options='<?php echo json_encode ( $item_options ); ?>'>
-						
-						<div class="fw-query-item"></div>
+		<div id="variable-grid" class="py-6" data-args='<?php echo json_encode ( $variables_args ); ?>'>
+			
+			<div id="" class="query-container ">
```
Comment (xfrenette-crim) [inline fw-child/template/variable/query.php:74]:
```suggestion
			<div class="query-container">
```

File: fw-child/languages/cdc/fr_CA.po [hunk 16]
```diff
@@ -972,6 +971,14 @@ msgstr ""
...
 msgstr ""
 
+#: template/learn/query.php:28
+msgid "Title (A–Z)"
+msgstr "Titre (A à Z)"
+
+#: template/learn/query.php:29
+msgid "Title (Z–A)"
+msgstr "Titre (Z à A)"
+
```
Comment (xfrenette-crim) [inline fw-child/languages/cdc/fr_CA.po:981]: This file (and the .pot and .mo files) would need to be regenerated since those texts are not used anymore.

**Review by xfrenette-crim — CHANGES_REQUESTED** [review]

File: fw-child/resources/scss/child.scss [hunk 17]
```diff
@@ -552,25 +546,42 @@ $control-slider-handle-width: 2.75rem;
...
-  .tab-drawer-tabs {
-    padding-left: 0.5rem;
-
-    a {
-      // display: block;
+  // Accomodate the "Rétroaction" fr wording to fit in the tiny sidebar
+  [lang="fr"] & {
+    .control-bar-text-link {
+      @include media-breakpoint-only(sm) {
+        hyphens: none;
```
[the remark this answers is not in the document]
Comment (domcode-ew) [reply fw-child/resources/scss/child.scss:588]: Just trying to make it look ok on all different breakpoints. The hyphen does not apply on sm viewport with this rule. It avoids this:  
Capture d’écran, le 2024-08-22 à 14 02 54

File: fw-child/template/apps/query.php [hunk 18]
```diff
@@ -21,39 +21,36 @@
...
-					'post_type' => 'app',
-					'post_parent' => 0,
-					'post_status' => 'publish'
-				);
-				
-			?>
+			<ul id="sort-menu" class="fw-query-sort list-unstyled d-flex all-caps">
+				<li class="selected me-4" data-sort="date_desc"><i class="fas fa-arrow-up me-2"></i> <?php _e ( 'Newest First', 'cdc' ); ?></li>
+				<li class="" data-sort="date_asc"><i class="fas fa-arrow-down me-2"></i> <?php _e ( 'Oldest First', 'cdc' ); ?></li>
+			</ul>
```
[the remark this answers is not in the document]
Comment (domcode-ew) [reply fw-child/template/apps/query.php:30]: This has been taken car of in the ticket regarding the Apps page.

File: fw-child/template/apps/query.php (hunk 11 above)
Comment (domcode-ew) [reply fw-child/template/apps/query.php:47]: This has been taken car of in the ticket regarding the Apps page.

File: fw-child/template/learn/query.php (hunk 12 above)
Comment (domcode-ew) [reply fw-child/template/learn/query.php:84]: This has been taken car of in the ticket regarding the Apps page.

File: fw-child/resources/scss/_template.scss
[the remark this answers is not in the document]
Comment (domcode-ew) [reply fw-child/resources/scss/_template.scss:1]: Changes have been made in the CMS to accommodate the width of the new blocks using the block's settings.

File: fw-child/resources/scss/_pages.scss
[the remark this answers is not in the document]
Comment (xfrenette-crim) [reply fw-child/resources/scss/_pages.scss:1]: The "Download" page is not being displayed correctly (the vertical shadow line and the help message shown under the sidebar):

image

Same with the "Maps" page:

image

Also, it's not possible to click on the items in the sidebars (for example, in the Maps page, it's not possible to change the variable displayed).

**Review by xfrenette-crim — CHANGES_REQUESTED** [review]
The Maps and Download page still don't work correctly.

**Review by xfrenette-crim — APPROVED** [review]
Approved. Reminder: the `demo-2.0` branch may also have changed `cdc.pot`, so the branch should first be merged here, then all of the .pot, .po and .mo files be regenerated.
